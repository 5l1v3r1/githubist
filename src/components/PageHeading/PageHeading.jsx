// @flow

import React from 'react';

import styles from './PageHeading.scss';

type Props = {
  title: string,
  subtitle?: string,
};

const PageHeading = ({ title, subtitle }: Props) => (
  <section className={styles.container}>
    <h1 className={styles.title}>{title}</h1>
    {subtitle && <p className={styles.subtitle}>{subtitle}</p>}
  </section>
);

PageHeading.defaultProps = {
  subtitle: '',
};

export default PageHeading;
