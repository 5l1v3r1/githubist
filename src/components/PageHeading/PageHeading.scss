@import '../../theme/theme';

.container {
  padding: $section-gap / 2 0;
}

.title {
  margin: 0;
  color: $black-color;
  font-size: map-get($heading-sizes, h1);
  font-weight: 600;
}

.subtitle {
  margin: map-get($spacers, 2x-small) 0 0 0;
  color: $body-color;
  font-size: map-get($font-sizes, large);
}
