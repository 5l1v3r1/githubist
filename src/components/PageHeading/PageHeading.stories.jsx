// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import backgrounds from '@storybook/addon-backgrounds';

import PageHeading from './PageHeading';

storiesOf('PageHeading', module)
  .addDecorator(backgrounds([{ name: 'dark', value: '#2d2e33', default: true }]))
  .add('default', () => (
    <div style={{ backgroundColor: '#fff' }}>
      <PageHeading title="Repos" subtitle="Lorem ipsum dolar sit amet." />
    </div>
  ))
  .add('without subtitle', () => (
    <div style={{ backgroundColor: '#fff' }}>
      <PageHeading title="Repos" />
    </div>
  ));
