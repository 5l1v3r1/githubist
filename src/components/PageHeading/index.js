// @flow

import PageHeading from './PageHeading';

export default PageHeading;
