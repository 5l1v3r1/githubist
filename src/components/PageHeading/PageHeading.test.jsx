// @flow

import React from 'react';
import { shallow } from 'enzyme';

import PageHeading from './PageHeading';

it('renders correctly', () => {
  const wrapper = shallow(<PageHeading title="Repos" subtitle="Lorem ipsum dolar sit amet." />);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without subtitle', () => {
  const wrapper = shallow(<PageHeading title="Repos" />);

  expect(wrapper).toMatchSnapshot();
});
