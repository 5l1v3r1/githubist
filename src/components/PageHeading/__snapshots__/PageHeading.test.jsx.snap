// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <h1>
    Repos
  </h1>
  <p>
    Lorem ipsum dolar sit amet.
  </p>
</section>
`;

exports[`renders correctly without subtitle 1`] = `
<section>
  <h1>
    Repos
  </h1>
</section>
`;
